### DISCUSSION TITLE
Erro na formatação das tabelas do banco de Dados

### DISCUSSION THREAD

**Discussion by laertealmeida** [viniciussanchez/dataset-serialize#214] [General]
Meu amigo, na versão antiga do Delphi eu conseguia que o Serialize formatase corretamente meu retorno das tabelas mas fiz a atualização para o Delphi 11.3 baixei as novas atualizações do RESTRequest4D e do Serialize e  usei os mesmos procedimente e não funciona. Os testes que fiz com o postman aparecem os retornos normais mas quando vou fazer o mesmo com o RESTRequest4D os dados não vem. Exemplo: Quando algum campo esta: id_usuario. No postman aparece normal mas quando chamo através do RESTRequest4D os dados não vem. E se altero o campo para idusuario (tudo junto), funciona. Os retornos vem normal através do RESTRequest4D.   O que será que estou fazendo de errado? Se poder ma ajudar agradeço. 

uses
 DataSet.Serialize,

procedure Tdm.DataModuleCreate(Sender: TObject);
begin
   TDataSetSerializeConfig.GetInstance.CaseNameDefinition := TCaseNameDefinition.cndLower;
   TDataSetSerializeConfig.GetInstance.Import.DecimalSeparator := '.';

    Conn.Connected := true;
end;

****************************

uses
RESTRequest4D, DataSet.Serialize.Adapter.RESTRequest4D

procedure TDmAgenda.BuscarEventos(mes: string);
var
    Resp: IResponse;
    json: TJsonObject;
    DiaSemana : String;
begin
    try
        TabAgenda.FieldDefs.Clear;

         resp := TRequest.New.BaseURL(DmGlobal.BaseURL)
                    .Resource('eventos')
                    .Accept('application/json')
                    .AddParam('mes', mes)
                    .Adapters(TDataSetSerializeAdapter.New(TabAgenda))
                    .Get;

          
        if Resp.StatusCode <> 200 then
            raise Exception.Create(RespMessage(Resp));    }

    finally
        json.DisposeOf;
    end;

end;

**Comment by viniciussanchez** [comment]
Bom dia.. aparentemente está correto essa parte da requisição, o que precisava fazer você já fez que é configurar o `CaseNameDefinition`.
Talvez o erro esteja na API agora não está? Talvez a API também usa o DataSet-Serialize e lá você não está configurando o `CaseNameDefinition`. Não sei... mas aparentemente seu código acima está normal.

**Reply by laertealmeida** [reply]
Eu até desistalei o Delphi pensando que era a instalação. 
Alguma sugestão ? Será que é banco de dados? O banco é postgre. 

O que me intriga que o meu backend com a versão antiga do Delphi funciona normal

**Reply by viniciussanchez** [reply]
Não... no seu backend você usa o dataset-serialize tambem? Verificou se lá ele também está configurando o CaseNameDefinition?

**Reply by laertealmeida** [reply]
sim esta😔

**Reply by laertealmeida** [reply]
Para os que passarem por este mesmo problema eu incluir um data Modulo no app cliente e adicionei a linha de codigo TDataSetSerializeConfig.GetInstance.CaseNameDefinition := TCaseNameDefinition.cndLower; no create e deu certo. 

Tanto no Backend como no aplicativo precisei usar o CaseNameDefinition.
